### ISSUE TITLE
[Feature request] [it's been a long time] real osu! game type in Stepmania

### ISSUE THREAD

**Issue by JOELwindows7** [stepmania/stepmania#1032]
Just a short appeal. why do we have to stuck on a just simply mainstream coding?
uh sorry anyway,

if you know osu!, you know what's the click the beat is.
on the other hand, they also simulates O2Jam which marketed as osu!mania. this game mode can be simulated to DDR or now reffered as Stepmania by simply make the skin of that for the mania mode.

We don't we revenge? Until the Linux now allowed the mouse click event or you figured out how to code mouse click event for Unix, How about the osu! game type __INSIDE__ Stepmania, __NOT__ Stepmania inside osu! (already been)

Yes,

__osu! gametype in Stepmania__
-----------------------------------------------

I was complained when tried to Google osu! in Stepmania then got Stepmania in osu!.

Simple, no need explanation otherwise learn from www.osu.ppy.sh
in short, a click the circle as the ring closes to it based on the rhythm.

You need to make
* Gameplay of course
* Editor for it inside the game, which is very different than usual
* etc..

Sorry, did I too much talking than contributed to? large apologise :(

**Comment by JOELwindows7** [comment]
Oh please don't disagree with this problem, otherwise Stomp ~~won't be open source anymore~~ , ummmmmm

**Comment by Kaiveran** [comment]
This is something that will definitely have to wait until SM6, where we will have a proper touch interface to work with.

It would also require an _entirely_ new chart format to encode position and movement (which I'm actually working on, but it's only in infant stages right now.) Even then, energies are likely to be focused on more like CROSSxBEATS and ReRave.

Although, come to think of it, a ReRave mode with position scoring turned off (and arc/bezier/spline capabilities for follows) would play almost exactly like Osu. And giving it an Osu-like look could be achieved with noteskin trickery. But in any case, it's not coming anytime soon.

**Comment by JOELwindows7** [comment]
@Kaiveran Yes, all I need is the touch and mouse support.
I know it's very hard challenge so it's not going to be important as well

**Comment by JOELwindows7** [comment]
@Kaiveran , did you see on a sentences editor? Yes I know it would be very different and requires drastic change to the chart format.

**Comment by 1033Forest** [comment]
ReRave mode sounds good. This way people can play it on their touch computers, and not have to buy an iPhone or iPad just to play the iOS version of it
